### PR TITLE
Add support for SignalTask.

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan <smahajan@roblox.com>

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
This patch can be testing using the following example job.

```
job "example" {
  datacenters = ["dc1"]

  group "cache" {
    task "redis" {
      driver = "podman"

      config {
        image = "docker://shm32/signal_handler:1.0"

        port_map {
          db = 6379
        }
      }

      resources {
        cpu    = 500
        memory = 256

        network {
          mbits = 10
          port  "db"  {}
        }
      }
    }
  }
}
```

```
$ nomad job run example.nomad
```

```
<Second terminal>
$ nomad alloc signal -s SIGHUP 61036c06

<First terminal>
$ nomad alloc logs -f 61036c06
2020-08-18T23:12:30.081746628+00:00 stdout F Hello signal_handler. My PID is 1
2020-08-18T23:14:05.132489761+00:00 stdout F Received SIGHUP
```

```
<Second terminal>
$ nomad alloc signal -s SIGUSR1 61036c06

<First Terminal>
$ nomad alloc logs -f 61036c06
2020-08-18T23:12:30.081746628+00:00 stdout F Hello signal_handler. My PID is 1
2020-08-18T23:14:05.132489761+00:00 stdout F Received SIGHUP
2020-08-18T23:15:25.163521867+00:00 stdout F Received SIGUSR1
```

File: driver.go [hunk 1]
```diff
@@ -668,7 +670,21 @@ func (d *Driver) TaskEvents(ctx context.Context) (<-chan *drivers.TaskEvent, err
...
+	}
+
+	// The given signal will be forwarded to the target taskID.
+	// Please checkout https://github.com/hashicorp/consul-template/blob/master/signals/signals_unix.go
+	// for a list of supported signals.
+	sig := os.Interrupt
+	if s, ok := signals.SignalLookup[signal]; ok {
+		sig = s
+	} else {
+		d.logger.Warn("unknown signal to send to task, using SIGINT instead", "signal", signal, "task_id", handle.taskConfig.ID)
```
[the remark this answers is not in the document]
Comment (shishir-a412ed) [reply driver.go]: Fixed.

**Comment by shishir-a412ed** [comment]
@towe75 

```
[vagrant@fedora30 example]$ nomad alloc signal -s SIGHUP 08bef401
<On second terminal>
[vagrant@fedora30 example]$ nomad alloc logs -f -job example
2020-08-24T20:09:41.400344873+00:00 stdout F Hello signal_handler. My PID is 1
2020-08-24T20:10:46.421778893+00:00 stdout F Received SIGHUP


[vagrant@fedora30 example]$ nomad alloc signal -s HUP 08bef401
Error signalling allocation: Unexpected response code: 500 (1 error occurred:
	* Failed to signal task: redis, err: rpc error: code = Unknown desc = Error in looking up signal: HUP

)
```

**Comment by shishir-a412ed** [comment]
@towe75 I am contemplating on the error message. Which one do you think would be better?

1) `Error in looking up signal: HUP`

OR 

2) `Invalid signal: HUP`

**Comment by towe75** [comment]
@shishir-a412ed I would go for the second one. Less words are usually better ;-)
But i don't mind if you keep first one.

**Comment by shishir-a412ed** [comment]
@towe75 Sounds good! I was leaning towards (2) as well. 
Already made the same change in our containerd task driver: https://github.com/Roblox/nomad-driver-containerd/pull/34

Let me update the PR.

**Comment by shishir-a412ed** [comment]
@towe75 Updated the error message. PTAL.

**Review by towe75 — APPROVED** [review]
LGTM 🚀

**Pull request merged**